File: src/DesignPatterns/src/Java/002-GOF-Patterns/003-BehavioralPatterns/009-Visitor/Visitor/src/org/csystem/app/ComputerPartVisitor.java
package org.csystem.app;

public class ComputerPartVisitor implements IComputerVisitor{
    @Override
    public void visit(Mouse mouse)
    {
        System.out.println("Mouse visit");
    }

    @Override
    public void visit(Keyboard keyboard)
    {
        System.out.println("Keyboard visit");
    }

    @Override
    public void visit(Computer keyboard)
    {
        System.out.println("Computer visit");
    }
}

File: src/DesignPatterns/src/C++/002-GOFPatterns/003-BehavioralPatterns/009-Visitor/ComputerVisitor/ComputerPartVisitor.cpp
#include "ComputerPartVisitor.h"

#include <iostream>

using namespace std;

void ComputerPartVisitor::Visit(Mouse *m)
{
	cout << "Mouse" << endl;
}

void ComputerPartVisitor::Visit(Keyboard *k)
{
	cout << "Keyboard" << endl;
}

void ComputerPartVisitor::Visit(Monitor *m)
{
	cout << "Monitor" << endl;
}

void ComputerPartVisitor::Visit(Computer *c)
{
	cout << "Computer" << endl;
}